presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java

```diff
@@ -358,11 +379,29 @@ public boolean supportClientTransaction()
                 .collect(toImmutableList());
     }
 
+    /**
+     * Extracts a specific header PRESTO_SESSION
+     *
+     * @param headers headers a MultivaluedMap containing all request headers.
+     *
+     * @return a Map with the {@link PRESTO_SESSION} properties
+     */
     private static Map<String, String> parseSessionHeaders(MultivaluedMap<String, String> headers)
     {
         return parseProperty(headers, PRESTO_SESSION);
     }
 
+    /**
+     * Checks if the received headers matches the defined headers pattern.
```

Comment:
This is only true for the PRESTO_ROLE header. The method is specific to this header.